config/datanode.example.toml

```diff
@@ -101,3 +101,17 @@ parallel_scan_channel_size = 32
 # [logging]
 # dir = "/tmp/greptimedb/logs"
 # level = "info"
+
+# datanode export the metrics generated by itself
+# encoded to Prometheus remote-write format
+# and send to Prometheus remote-write compatible receiver (e.g. send to `greptimedb` itself)
+# This is only used for `greptimedb` to export its own metric internally. Please see `logging` option for normal export of metric.
+# [system_metric]
```

Comment:
What about `[export_metrics]`? I think it's more precise.

Reply:
Agree.

For @waynexia saying:
> Standalone and frontend should be able to avoid the network stack by invoking themselves.
>
> And prefer to config endpoint, database, extra header etc separately.

I recommend strictly distinguishing these two export methods in the configuration, using different sub-configs to handle them:

```
[export_metrics.remote_write]
endpoint = "http://127.0.0.1:4000/v1/prometheus/write?db=information_schema"

[export_metrics.self_inject]
db = "information_schema"
```

Reply:
For now we lack the ability to find `frontend` http endpoint in `metasrv` and `datanode`, for `metasrv` and `datanode` we still need manually pass `frontend` http endpoint, so maybe the following config is more unify and easy to understand to user.

```toml
[export_metrics]
enable = true
endpoint = "127.0.0.1:4000"
db = "information_schema"
write_interval = "30s"
headers = ["Auth....", ]
```

Reply:
> for metasrv and datanode we still need manually pass frontend http endpoint

Then you can config in this flavor:

```
[export_metrics]
enable = true
interval = "30s"

[export_metrics.remote_write]
endpoint = "http://frontend_addr:4000/v1/prometheus/write?db=my_db"
headers = []
```